src/execution/compiler/operator/hash_aggregation_translator.cpp

```diff
@@ -29,6 +29,19 @@ HashAggregationTranslator::HashAggregationTranslator(const planner::AggregatePla
   TERRIER_ASSERT(plan.GetAggregateStrategyType() == planner::AggregateStrategyType::HASH,
                  "Expected hash-based aggregation plan node");
   TERRIER_ASSERT(plan.GetChildrenSize() == 1, "Hash aggregations should only have one child");
+  // TODO(kunal&ricky):
+  // We need to modify many of the setup/initialize/teardown logic to add the additional agg_distinct_ht.
+  // A agg_distinct_ht will be a global mapping [distinct aggregate value -> dummy payload],
+  // for each aggregate term that has a distinct predicate.
+  // The agg_distinc_ht should have the same setup/teardown logic like the current global_agg_ht_. But we need to
+  // keep track of multiple hash tables, or we need to have a complex key (aggregate_term_id, tuple_val) to
```

Comment:
Yes, you will need to keep track of multiple hash tables. That is what Amadou's code does.

Reply:
If we have some sort of id to distinguish different aggregate terms, do we still need to keep track of multiple hash tables?

Reply:
I think you are right. I need to keep track of multiple hash tables, each for one aggregate, since different aggregates might have different return types.